backend/authBackend.js

```diff
@@ -1,65 +1,54 @@
-// JWT is our primary identification mechanism
-import * as jwt from 'jsonwebtoken';
+import { getCurrentUser } from '@reshuffle/server-function';
 
-// used to validate incoming Google profiles from client
-import { OAuth2Client } from 'google-auth-library';
+const { VALID_HOSTED_DOMAINS } = process.env;
+const domains = VALID_HOSTED_DOMAINS.split(' ');
 
-// we depend on both OAUTH_CLIENT_ID and a SECURE
-// HMAC which is used to sign our tokens
-//
-// THE HMAC CANNOT BE LEAKED!!!!!!!!!!!!!!
-const {
-  REACT_APP_OAUTH_CLIENT_ID,
-  JWT_HMAC_KEY,
-  REACT_APP_VALID_HOSTED_DOMAIN,
-} = process.env;
+/**
+* validates if email is the right VALID_HOSTED_DOMAIN
+* setting getCurrentUser to false since I'm calling this function in the front end and I don't want
+* want the CRA to crash for not being logged in if user hits this function.
+/** @expose */
+export async function checkIfValidDomain() {
+  const profile = getCurrentUser(false);
 
-const client = new OAuth2Client(REACT_APP_OAUTH_CLIENT_ID);
+  if (profile === undefined) {
+    return false;
+  }
 
-/**
- * Makes the necessary requests to Google auth servers
- * to verify that the client user is actually who they
- * say they are.
- *
- * @param { string } token - idToken returned by clientside auth
- */
-async function verifyGoogleUser(token) {
-  const ticket = await client.verifyIdToken({
-    idToken: token,
-    audience: REACT_APP_OAUTH_CLIENT_ID,
-  });
-  return ticket.getPayload();
+  const emails = profile.emails.map((email) => {
+    return email.value
+  })
```

Comment:
We prefer to avoid (ASI) automatic semicolon insertion in JavaScript because it can have painful unintended consequences.  It's probably OK if you configure lint properly and do it consistently throughout the project, otherwise I'd like to avoid it in this project too.
(Also there's a lint issue right here, we should be failing these 2 lines precisely because they _do_ rely on ASI.)